Build brpc failed

**Describe the bug**
Build brpc failed

**Environment**
- OS: [ubuntu 20]
- Compiler: revision

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install brpc

Repro code when

**Expected behavior**
A clear and concise description of what you expected to happen.

**Failure logs**
```shell
-- Applying patch fix_boost_ptr.patch
-- Applying patch fix_thrift.patch
-- Applying patch /home/zreal/research/vcpkg/downloads/apache-incubator-brpc-1679.diff
-- Using source at /home/zreal/research/vcpkg/buildtrees/brpc/src/a1b8d7f990-fc2e8574c2.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: /usr/local/bin/cmake --build . --config Debug --target install -- -v -j9
    Working Directory: /home/zreal/research/vcpkg/buildtrees/brpc/x64-linux-dbg
    See logs for more information:
      /home/zreal/research/vcpkg/buildtrees/brpc/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:111 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/brpc/portfile.cmake:27 (vcpkg_cmake_install)
  scripts/ports.cmake:145 (include)


Error: Building package brpc:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+brpc
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[brpc]+Build+error
including:
  package: brpc[core]:x64-linux -> 0.9.7#7
    vcpkg-tool version: 2022-03-25-d9702d9d89b9468c32efc3f18fb84c3021d7dbae
    vcpkg-scripts version: 9d9a6f486 2022-03-30 (15 hours ago)

Additionally, attach any relevant sections from the log files above.

```

**Additional context**
Add any other context about the problem here.

Complete Log Info
```shell
zreal@Zreal-PC:~/research/vcpkg$ ./vcpkg install brpc
Computing installation plan...
The following packages will be built and installed:
  * boost-algorithm[core]:x64-linux -> 1.78.0
  * boost-align[core]:x64-linux -> 1.78.0
  * boost-array[core]:x64-linux -> 1.78.0
  * boost-assert[core]:x64-linux -> 1.78.0
  * boost-atomic[core]:x64-linux -> 1.78.0
  * boost-bind[core]:x64-linux -> 1.78.0
  * boost-build[core]:x64-linux -> 1.78.0#1
  * boost-chrono[core]:x64-linux -> 1.78.0
  * boost-concept-check[core]:x64-linux -> 1.78.0
  * boost-config[core]:x64-linux -> 1.78.0
  * boost-container[core]:x64-linux -> 1.78.0
  * boost-container-hash[core]:x64-linux -> 1.78.0
  * boost-conversion[core]:x64-linux -> 1.78.0
  * boost-core[core]:x64-linux -> 1.78.0
  * boost-date-time[core]:x64-linux -> 1.78.0
  * boost-detail[core]:x64-linux -> 1.78.0
  * boost-exception[core]:x64-linux -> 1.78.0
  * boost-function[core]:x64-linux -> 1.78.0
  * boost-function-types[core]:x64-linux -> 1.78.0
  * boost-fusion[core]:x64-linux -> 1.78.0
  * boost-integer[core]:x64-linux -> 1.78.0
  * boost-intrusive[core]:x64-linux -> 1.78.0
  * boost-io[core]:x64-linux -> 1.78.0
  * boost-iterator[core]:x64-linux -> 1.78.0
  * boost-lexical-cast[core]:x64-linux -> 1.78.0
  * boost-locale[core]:x64-linux -> 1.78.0
  * boost-modular-build-helper[core]:x64-linux -> 1.78.0#3
  * boost-move[core]:x64-linux -> 1.78.0
  * boost-mp11[core]:x64-linux -> 1.78.0
  * boost-mpl[core]:x64-linux -> 1.78.0
  * boost-numeric-conversion[core]:x64-linux -> 1.78.0
  * boost-optional[core]:x64-linux -> 1.78.0
  * boost-predef[core]:x64-linux -> 1.78.0
  * boost-preprocessor[core]:x64-linux -> 1.78.0
  * boost-range[core]:x64-linux -> 1.78.0
  * boost-ratio[core]:x64-linux -> 1.78.0
  * boost-rational[core]:x64-linux -> 1.78.0
  * boost-regex[core]:x64-linux -> 1.78.0
  * boost-scope-exit[core]:x64-linux -> 1.78.0
  * boost-smart-ptr[core]:x64-linux -> 1.78.0
  * boost-static-assert[core]:x64-linux -> 1.78.0
  * boost-system[core]:x64-linux -> 1.78.0
  * boost-thread[core]:x64-linux -> 1.78.0
  * boost-throw-exception[core]:x64-linux -> 1.78.0
  * boost-tokenizer[core]:x64-linux -> 1.78.0
  * boost-tuple[core]:x64-linux -> 1.78.0
  * boost-type-index[core]:x64-linux -> 1.78.0
  * boost-type-traits[core]:x64-linux -> 1.78.0
  * boost-typeof[core]:x64-linux -> 1.78.0
  * boost-uninstall[core]:x64-linux -> 1.78.0
  * boost-unordered[core]:x64-linux -> 1.78.0
  * boost-utility[core]:x64-linux -> 1.78.0
  * boost-variant2[core]:x64-linux -> 1.78.0
  * boost-vcpkg-helpers[core]:x64-linux -> 1.78.0#1
  * boost-winapi[core]:x64-linux -> 1.78.0
    brpc[core]:x64-linux -> 0.9.7#7
  * gflags[core]:x64-linux -> 2.2.2#5
  * glog[core]:x64-linux -> 0.4.0#4
  * leveldb[core]:x64-linux -> 1.22#5
  * libevent[core,thread]:x64-linux -> 2.1.12#6
  * libiconv[core]:x64-linux -> 1.16#12
  * openssl[core]:x64-linux -> 1.1.1n#1
  * protobuf[core,zlib]:x64-linux -> 3.19.4
  * thrift[core]:x64-linux -> 0.13.0#2
  * vcpkg-cmake[core]:x64-linux -> 2022-02-14
  * vcpkg-cmake-config[core]:x64-linux -> 2022-02-06
  * zlib[core]:x64-linux -> 1.2.12
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Restored 66 packages from /home/zreal/.cache/vcpkg/archives in 4.016 s. Use --debug to see more details.
Starting package 1/67: boost-uninstall:x64-linux
Installing package boost-uninstall[core]:x64-linux...
Elapsed time for package boost-uninstall:x64-linux: 34.47 ms
Starting package 2/67: boost-vcpkg-helpers:x64-linux
Installing package boost-vcpkg-helpers[core]:x64-linux...
Elapsed time for package boost-vcpkg-helpers:x64-linux: 3.457 ms
Starting package 3/67: boost-config:x64-linux
Installing package boost-config[core]:x64-linux...
Elapsed time for package boost-config:x64-linux: 468.7 ms
Starting package 4/67: boost-assert:x64-linux
Installing package boost-assert[core]:x64-linux...
Elapsed time for package boost-assert:x64-linux: 14.9 ms
Starting package 5/67: boost-static-assert:x64-linux
Installing package boost-static-assert[core]:x64-linux...
Elapsed time for package boost-static-assert:x64-linux: 5.807 ms
Starting package 6/67: boost-throw-exception:x64-linux
Installing package boost-throw-exception[core]:x64-linux...
Elapsed time for package boost-throw-exception:x64-linux: 55.03 ms
Starting package 7/67: boost-core:x64-linux
Installing package boost-core[core]:x64-linux...
Elapsed time for package boost-core:x64-linux: 86.1 ms
Starting package 8/67: boost-array:x64-linux
Installing package boost-array[core]:x64-linux...
Elapsed time for package boost-array:x64-linux: 33.9 ms
Starting package 9/67: boost-bind:x64-linux
Installing package boost-bind[core]:x64-linux...
Elapsed time for package boost-bind:x64-linux: 291.5 ms
Starting package 10/67: boost-preprocessor:x64-linux
Installing package boost-preprocessor[core]:x64-linux...
Elapsed time for package boost-preprocessor:x64-linux: 2.433 s
Starting package 11/67: boost-type-traits:x64-linux
Installing package boost-type-traits[core]:x64-linux...
Elapsed time for package boost-type-traits:x64-linux: 1.479 s
Starting package 12/67: boost-concept-check:x64-linux
Installing package boost-concept-check[core]:x64-linux...
Elapsed time for package boost-concept-check:x64-linux: 100.5 ms
Starting package 13/67: boost-build:x64-linux
Installing package boost-build[core]:x64-linux...
Elapsed time for package boost-build:x64-linux: 1.26 s
Starting package 14/67: vcpkg-cmake:x64-linux
Installing package vcpkg-cmake[core]:x64-linux...
Elapsed time for package vcpkg-cmake:x64-linux: 1.554 ms
Starting package 15/67: boost-modular-build-helper:x64-linux
Installing package boost-modular-build-helper[core]:x64-linux...
Elapsed time for package boost-modular-build-helper:x64-linux: 1.526 ms
Starting package 16/67: boost-predef:x64-linux
Installing package boost-predef[core]:x64-linux...
Elapsed time for package boost-predef:x64-linux: 8.357 ms
Starting package 17/67: boost-winapi:x64-linux
Installing package boost-winapi[core]:x64-linux...
Elapsed time for package boost-winapi:x64-linux: 9.486 ms
Starting package 18/67: boost-move:x64-linux
Installing package boost-move[core]:x64-linux...
Elapsed time for package boost-move:x64-linux: 3.777 ms
Starting package 19/67: boost-smart-ptr:x64-linux
Installing package boost-smart-ptr[core]:x64-linux...
Elapsed time for package boost-smart-ptr:x64-linux: 7.457 ms
Starting package 20/67: boost-tuple:x64-linux
Installing package boost-tuple[core]:x64-linux...
Elapsed time for package boost-tuple:x64-linux: 3.164 ms
Starting package 21/67: boost-exception:x64-linux
Installing package boost-exception[core]:x64-linux...
Elapsed time for package boost-exception:x64-linux: 8.353 ms
Starting package 22/67: boost-integer:x64-linux
Installing package boost-integer[core]:x64-linux...
Elapsed time for package boost-integer:x64-linux: 4.029 ms
Starting package 23/67: boost-detail:x64-linux
Installing package boost-detail[core]:x64-linux...
Elapsed time for package boost-detail:x64-linux: 7.12 ms
Starting package 24/67: boost-container-hash:x64-linux
Installing package boost-container-hash[core]:x64-linux...
Elapsed time for package boost-container-hash:x64-linux: 3.858 ms
Starting package 25/67: boost-type-index:x64-linux
Installing package boost-type-index[core]:x64-linux...
Elapsed time for package boost-type-index:x64-linux: 4.239 ms
Starting package 26/67: boost-typeof:x64-linux
Installing package boost-typeof[core]:x64-linux...
Elapsed time for package boost-typeof:x64-linux: 66 ms
Starting package 27/67: boost-function:x64-linux
Installing package boost-function[core]:x64-linux...
Elapsed time for package boost-function:x64-linux: 89.06 ms
Starting package 28/67: boost-conversion:x64-linux
Installing package boost-conversion[core]:x64-linux...
Elapsed time for package boost-conversion:x64-linux: 7.455 ms
Starting package 29/67: boost-io:x64-linux
Installing package boost-io[core]:x64-linux...
Elapsed time for package boost-io:x64-linux: 8.88 ms
Starting package 30/67: boost-utility:x64-linux
Installing package boost-utility[core]:x64-linux...
Elapsed time for package boost-utility:x64-linux: 16.02 ms
Starting package 31/67: boost-mpl:x64-linux
Installing package boost-mpl[core]:x64-linux...
Elapsed time for package boost-mpl:x64-linux: 189.8 ms
Starting package 32/67: boost-function-types:x64-linux
Installing package boost-function-types[core]:x64-linux...
Elapsed time for package boost-function-types:x64-linux: 6.25 ms
Starting package 33/67: boost-fusion:x64-linux
Installing package boost-fusion[core]:x64-linux...
Elapsed time for package boost-fusion:x64-linux: 527.3 ms
Starting package 34/67: boost-optional:x64-linux
Installing package boost-optional[core]:x64-linux...
Elapsed time for package boost-optional:x64-linux: 20.4 ms
Starting package 35/67: boost-iterator:x64-linux
Installing package boost-iterator[core]:x64-linux...
Elapsed time for package boost-iterator:x64-linux: 30.21 ms
Starting package 36/67: boost-regex:x64-linux
Installing package boost-regex[core]:x64-linux...
Elapsed time for package boost-regex:x64-linux: 214.5 ms
Starting package 37/67: boost-range:x64-linux
Installing package boost-range[core]:x64-linux...
Elapsed time for package boost-range:x64-linux: 104.4 ms
Starting package 38/67: boost-intrusive:x64-linux
Installing package boost-intrusive[core]:x64-linux...
Elapsed time for package boost-intrusive:x64-linux: 80.16 ms
Starting package 39/67: boost-container:x64-linux
Installing package boost-container[core]:x64-linux...
Elapsed time for package boost-container:x64-linux: 189.6 ms
Starting package 40/67: boost-unordered:x64-linux
Installing package boost-unordered[core]:x64-linux...
Elapsed time for package boost-unordered:x64-linux: 22.4 ms
Starting package 41/67: boost-algorithm:x64-linux
Installing package boost-algorithm[core]:x64-linux...
Elapsed time for package boost-algorithm:x64-linux: 36.1 ms
Starting package 42/67: boost-align:x64-linux
Installing package boost-align[core]:x64-linux...
Elapsed time for package boost-align:x64-linux: 26.8 ms
Starting package 43/67: boost-atomic:x64-linux
Installing package boost-atomic[core]:x64-linux...
Elapsed time for package boost-atomic:x64-linux: 97.82 ms
Starting package 44/67: boost-rational:x64-linux
Installing package boost-rational[core]:x64-linux...
Elapsed time for package boost-rational:x64-linux: 12.92 ms
Starting package 45/67: boost-ratio:x64-linux
Installing package boost-ratio[core]:x64-linux...
Elapsed time for package boost-ratio:x64-linux: 19.44 ms
Starting package 46/67: boost-mp11:x64-linux
Installing package boost-mp11[core]:x64-linux...
Elapsed time for package boost-mp11:x64-linux: 7.057 ms
Starting package 47/67: boost-variant2:x64-linux
Installing package boost-variant2[core]:x64-linux...
Elapsed time for package boost-variant2:x64-linux: 3.324 ms
Starting package 48/67: boost-system:x64-linux
Installing package boost-system[core]:x64-linux...
Elapsed time for package boost-system:x64-linux: 38.08 ms
Starting package 49/67: boost-chrono:x64-linux
Installing package boost-chrono[core]:x64-linux...
Elapsed time for package boost-chrono:x64-linux: 44.85 ms
Starting package 50/67: boost-numeric-conversion:x64-linux
Installing package boost-numeric-conversion[core]:x64-linux...
Elapsed time for package boost-numeric-conversion:x64-linux: 33.8 ms
Starting package 51/67: boost-lexical-cast:x64-linux
Installing package boost-lexical-cast[core]:x64-linux...
Elapsed time for package boost-lexical-cast:x64-linux: 93.43 ms
Starting package 52/67: boost-tokenizer:x64-linux
Installing package boost-tokenizer[core]:x64-linux...
Elapsed time for package boost-tokenizer:x64-linux: 14.1 ms
Starting package 53/67: boost-date-time:x64-linux
Installing package boost-date-time[core]:x64-linux...
Elapsed time for package boost-date-time:x64-linux: 54.2 ms
Starting package 54/67: boost-thread:x64-linux
Installing package boost-thread[core]:x64-linux...
Elapsed time for package boost-thread:x64-linux: 104.6 ms
Starting package 55/67: libiconv:x64-linux
Installing package libiconv[core]:x64-linux...
Elapsed time for package libiconv:x64-linux: 11.89 ms
Starting package 56/67: boost-locale:x64-linux
Installing package boost-locale[core]:x64-linux...
Elapsed time for package boost-locale:x64-linux: 318.8 ms
Starting package 57/67: boost-scope-exit:x64-linux
Installing package boost-scope-exit[core]:x64-linux...
Elapsed time for package boost-scope-exit:x64-linux: 37.79 ms
Starting package 58/67: gflags:x64-linux
Installing package gflags[core]:x64-linux...
Elapsed time for package gflags:x64-linux: 35.17 ms
Starting package 59/67: glog:x64-linux
Installing package glog[core]:x64-linux...
Elapsed time for package glog:x64-linux: 44.43 ms
Starting package 60/67: vcpkg-cmake-config:x64-linux
Installing package vcpkg-cmake-config[core]:x64-linux...
Elapsed time for package vcpkg-cmake-config:x64-linux: 6.24 ms
Starting package 61/67: leveldb:x64-linux
Installing package leveldb[core]:x64-linux...
Elapsed time for package leveldb:x64-linux: 116.5 ms
Starting package 62/67: openssl:x64-linux
Installing package openssl[core]:x64-linux...
Elapsed time for package openssl:x64-linux: 430.3 ms
Starting package 63/67: zlib:x64-linux
Installing package zlib[core]:x64-linux...
Elapsed time for package zlib:x64-linux: 80.77 ms
Starting package 64/67: protobuf:x64-linux
Installing package protobuf[core,zlib]:x64-linux...
Elapsed time for package protobuf:x64-linux: 1.928 s
Starting package 65/67: libevent:x64-linux
Installing package libevent[core,thread]:x64-linux...
Elapsed time for package libevent:x64-linux: 96.92 ms
Starting package 66/67: thrift:x64-linux
Installing package thrift[core]:x64-linux...
Elapsed time for package thrift:x64-linux: 202.7 ms
Starting package 67/67: brpc:x64-linux
Building package brpc[core]:x64-linux...
-- Downloading https://patch-diff.githubusercontent.com/raw/apache/incubator-brpc/pull/1679.diff -> apache-incubator-brpc-1679.diff...
-- Downloading https://github.com/apache/incubator-brpc/archive/1d6510aa50075cade5ed539ee09a11a1b8d7f990.tar.gz -> apache-incubator-brpc-1d6510aa50075cade5ed539ee09a11a1b8d7f990.tar.gz...
-- Extracting source /home/zreal/research/vcpkg/downloads/apache-incubator-brpc-1d6510aa50075cade5ed539ee09a11a1b8d7f990.tar.gz
-- Applying patch fix_boost_ptr.patch
-- Applying patch fix_thrift.patch
-- Applying patch /home/zreal/research/vcpkg/downloads/apache-incubator-brpc-1679.diff
-- Using source at /home/zreal/research/vcpkg/buildtrees/brpc/src/a1b8d7f990-fc2e8574c2.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: /usr/local/bin/cmake --build . --config Debug --target install -- -v -j9
    Working Directory: /home/zreal/research/vcpkg/buildtrees/brpc/x64-linux-dbg
    See logs for more information:
      /home/zreal/research/vcpkg/buildtrees/brpc/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:111 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/brpc/portfile.cmake:27 (vcpkg_cmake_install)
  scripts/ports.cmake:145 (include)


Error: Building package brpc:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+brpc
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[brpc]+Build+error
including:
  package: brpc[core]:x64-linux -> 0.9.7#7
    vcpkg-tool version: 2022-03-25-d9702d9d89b9468c32efc3f18fb84c3021d7dbae
    vcpkg-scripts version: 9d9a6f486 2022-03-30 (15 hours ago)

Additionally, attach any relevant sections from the log files above.

```

--- a/ports/brpc/brpc-1783.diff
+++ b/ports/brpc/brpc-1783.diff
@@ -1,0 +1,592 @@
+diff --git a/BUILD.bazel b/BUILD.bazel
+index 11db84d02..7592a1862 100644
+--- a/BUILD.bazel
++++ b/BUILD.bazel
+@@ -56,7 +56,7 @@ config_setting(
+ 
+ COPTS = [
+     "-DBTHREAD_USE_FAST_PTHREAD_MUTEX",
+-    "-D__const__=",
++    "-D__const__=__unused__",
+     "-D_GNU_SOURCE",
+     "-DUSE_SYMBOLIZE",
+     "-DNO_TCMALLOC",
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 057695afc..bfc9fe4f0 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -108,7 +108,7 @@ set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} ${DEFINE_CLOCK_GETTIME} -DBRPC_WITH_GLOG
+ if(WITH_MESALINK)
+     set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} -DUSE_MESALINK")
+ endif()
+-set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} -DBTHREAD_USE_FAST_PTHREAD_MUTEX -D__const__= -D_GNU_SOURCE -DUSE_SYMBOLIZE -DNO_TCMALLOC -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DBRPC_REVISION=\\\"${BRPC_REVISION}\\\" -D__STRICT_ANSI__")
++set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} -DBTHREAD_USE_FAST_PTHREAD_MUTEX -D__const__=__unused__ -D_GNU_SOURCE -DUSE_SYMBOLIZE -DNO_TCMALLOC -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DBRPC_REVISION=\\\"${BRPC_REVISION}\\\" -D__STRICT_ANSI__")
+ set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} ${DEBUG_SYMBOL} ${THRIFT_CPP_FLAG}")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -O2 -pipe -Wall -W -fPIC -fstrict-aliasing -Wno-invalid-offsetof -Wno-unused-parameter -fno-omit-frame-pointer")
+ set(CMAKE_C_FLAGS "${CMAKE_CPP_FLAGS} -O2 -pipe -Wall -W -fPIC -fstrict-aliasing -Wno-unused-parameter -fno-omit-frame-pointer")
+diff --git a/Makefile b/Makefile
+index e2e1a6b2e..da506f7c5 100644
+--- a/Makefile
++++ b/Makefile
+@@ -20,10 +20,9 @@ include config.mk
+ 
+ # Notes on the flags:
+ # 1. Added -fno-omit-frame-pointer: perf/tcmalloc-profiler use frame pointers by default
+-# 2. Added -D__const__= : Avoid over-optimizations of TLS variables by GCC>=4.8
+-# 3. Removed -Werror: Not block compilation for non-vital warnings, especially when the
++# 2. Removed -Werror: Not block compilation for non-vital warnings, especially when the
+ #    code is tested on newer systems. If the code is used in production, add -Werror back
+-CPPFLAGS+=-DBTHREAD_USE_FAST_PTHREAD_MUTEX -D__const__= -D_GNU_SOURCE -DUSE_SYMBOLIZE -DNO_TCMALLOC -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DNDEBUG -DBRPC_REVISION=\"$(shell ./tools/get_brpc_revision.sh .)\"
++CPPFLAGS+=-DBTHREAD_USE_FAST_PTHREAD_MUTEX -D_GNU_SOURCE -DUSE_SYMBOLIZE -DNO_TCMALLOC -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DNDEBUG -DBRPC_REVISION=\"$(shell ./tools/get_brpc_revision.sh .)\"
+ CXXFLAGS=$(CPPFLAGS) -O2 -pipe -Wall -W -fPIC -fstrict-aliasing -Wno-invalid-offsetof -Wno-unused-parameter -fno-omit-frame-pointer -std=c++0x
+ CFLAGS=$(CPPFLAGS) -O2 -pipe -Wall -W -fPIC -fstrict-aliasing -Wno-unused-parameter -fno-omit-frame-pointer
+ DEBUG_CXXFLAGS = $(filter-out -DNDEBUG,$(CXXFLAGS)) -DUNIT_TEST -DBVAR_NOT_LINK_DEFAULT_VARIABLES
+diff --git a/config_brpc.sh b/config_brpc.sh
+index 842f100f9..cf04b7260 100755
+--- a/config_brpc.sh
++++ b/config_brpc.sh
+@@ -320,6 +320,10 @@ append_to_output "STATIC_LINKINGS=$STATIC_LINKINGS"
+ append_to_output "DYNAMIC_LINKINGS=$DYNAMIC_LINKINGS"
+ CPPFLAGS="-DBRPC_WITH_GLOG=$WITH_GLOG -DGFLAGS_NS=$GFLAGS_NS"
+ 
++# Avoid over-optimizations of TLS variables by GCC>=4.8
++# See: https://github.com/apache/incubator-brpc/issues/1693
++CPPFLAGS="${CPPFLAGS} -D__const__=__unused__"
++
+ if [ ! -z "$DEBUGSYMBOLS" ]; then
+     CPPFLAGS="${CPPFLAGS} $DEBUGSYMBOLS"
+ fi
+diff --git a/docs/cn/getting_started.md b/docs/cn/getting_started.md
+index 2602f5b41..56f80db3b 100644
+--- a/docs/cn/getting_started.md
++++ b/docs/cn/getting_started.md
+@@ -290,7 +290,7 @@ GCC7中over-aligned的问题暂时被禁止。
+ 
+ 使用其他版本的gcc可能会产生编译警告，请联系我们予以修复。
+ 
+-请在makefile中给cxxflags增加`-D__const__=`选项以避免[gcc4+中的errno问题](thread_local.md).
++请在makefile中给cxxflags增加`-D__const__=__unused__`选项以避免[gcc4+中的errno问题](thread_local.md).
+ 
+ ## Clang: 3.5-4.0
+ 
+diff --git a/docs/cn/thread_local.md b/docs/cn/thread_local.md
+index f8e1a491e..41e024749 100644
+--- a/docs/cn/thread_local.md
++++ b/docs/cn/thread_local.md
+@@ -57,9 +57,8 @@ Use *p ...                   -  still the errno of original pthread, undefined b
+ 
+ 严格地说这个问题不是gcc4导致的，而是glibc给__errno_location的签名不够准确，一个返回thread-local指针的函数依赖于段寄存器（TLS的一般实现方式），这怎么能算const呢？由于我们还未找到覆盖__errno_location的方法，所以这个问题目前实际的解决方法是：
+ 
+-**务必在直接或间接使用bthread的项目的gcc编译选项中添加`-D__const__=`，即把`__const__`定义为空，避免gcc4做相关优化。**
++**务必在直接或间接使用bthread的项目的gcc编译选项中添加`-D__const__=__unused__`，即把`__const__`定义为一个无副作用的属性，避免gcc4做相关优化。**
+ 
+-把`__const__`定义为空对程序其他部分的影响几乎为0。另外如果你没有**直接**使用errno（即你的项目中没有出现errno），或使用的是gcc
+-3.4，即使没有定义`-D__const__=`，程序的正确性也不会受影响，但为了防止未来可能的问题，我们强烈建议加上。
++把`__const__`定义为`__unused__`对程序其他部分的影响几乎为0。另外如果你没有**直接**使用errno（即你的项目中没有出现errno），或使用的是gcc 3.4，即使没有定义`-D__const__=__unused__`，程序的正确性也不会受影响，但为了防止未来可能的问题，我们强烈建议加上。
+ 
+-需要说明的是，和errno类似，pthread_self也有类似的问题，不过一般pthread_self除了打日志没有其他用途，影响面较小，在`-D__const__=`后pthread_self也会正常。
++需要说明的是，和errno类似，pthread_self也有类似的问题，不过一般pthread_self除了打日志没有其他用途，影响面较小，在`-D__const__=__unused__`后pthread_self也会正常。
+diff --git a/docs/en/getting_started.md b/docs/en/getting_started.md
+index c500f5807..4242f82a0 100644
+--- a/docs/en/getting_started.md
++++ b/docs/en/getting_started.md
+@@ -297,7 +297,7 @@ The over-aligned issues in GCC7 is suppressed temporarily now.
+ 
+ Using other versions of gcc may generate warnings, contact us to fix.
+ 
+-Adding `-D__const__=` to cxxflags in your makefiles is a must to avoid [errno issue in gcc4+](thread_local.md).
++Adding `-D__const__=__unused__` to cxxflags in your makefiles is a must to avoid [errno issue in gcc4+](thread_local.md).
+ 
+ ## Clang: 3.5-4.0
+ 
+diff --git a/example/BUILD b/example/BUILD
+index ee2c6ffd0..d688749d8 100644
+--- a/example/BUILD
++++ b/example/BUILD
+@@ -16,7 +16,7 @@
+ COPTS = [
+     "-D__STDC_FORMAT_MACROS",
+     "-DBTHREAD_USE_FAST_PTHREAD_MUTEX",
+-    "-D__const__=",
++    "-D__const__=__unused__",
+     "-D_GNU_SOURCE",
+     "-DUSE_SYMBOLIZE",
+     "-DNO_TCMALLOC",
+diff --git a/example/asynchronous_echo_c++/CMakeLists.txt b/example/asynchronous_echo_c++/CMakeLists.txt
+index 18fec20ee..4a118b19a 100644
+--- a/example/asynchronous_echo_c++/CMakeLists.txt
++++ b/example/asynchronous_echo_c++/CMakeLists.txt
+@@ -81,7 +81,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/auto_concurrency_limiter/CMakeLists.txt b/example/auto_concurrency_limiter/CMakeLists.txt
+index ef20bf0a7..88b784277 100644
+--- a/example/auto_concurrency_limiter/CMakeLists.txt
++++ b/example/auto_concurrency_limiter/CMakeLists.txt
+@@ -70,7 +70,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/backup_request_c++/CMakeLists.txt b/example/backup_request_c++/CMakeLists.txt
+index d247bc150..fc39ba3cd 100644
+--- a/example/backup_request_c++/CMakeLists.txt
++++ b/example/backup_request_c++/CMakeLists.txt
+@@ -81,7 +81,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/cancel_c++/CMakeLists.txt b/example/cancel_c++/CMakeLists.txt
+index 26f2581ba..ea611e03f 100644
+--- a/example/cancel_c++/CMakeLists.txt
++++ b/example/cancel_c++/CMakeLists.txt
+@@ -81,7 +81,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/cascade_echo_c++/CMakeLists.txt b/example/cascade_echo_c++/CMakeLists.txt
+index 24d9249ae..6ca2e25dc 100644
+--- a/example/cascade_echo_c++/CMakeLists.txt
++++ b/example/cascade_echo_c++/CMakeLists.txt
+@@ -80,7 +80,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/dynamic_partition_echo_c++/CMakeLists.txt b/example/dynamic_partition_echo_c++/CMakeLists.txt
+index 5a268a6c9..8df3ad62e 100644
+--- a/example/dynamic_partition_echo_c++/CMakeLists.txt
++++ b/example/dynamic_partition_echo_c++/CMakeLists.txt
+@@ -85,7 +85,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/echo_c++/CMakeLists.txt b/example/echo_c++/CMakeLists.txt
+index 4e6f5231d..d7babd7f7 100644
+--- a/example/echo_c++/CMakeLists.txt
++++ b/example/echo_c++/CMakeLists.txt
+@@ -81,7 +81,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/echo_c++/Makefile b/example/echo_c++/Makefile
+index 710cc7eda..fddde8cbc 100644
+--- a/example/echo_c++/Makefile
++++ b/example/echo_c++/Makefile
+@@ -20,8 +20,7 @@ BRPC_PATH=../..
+ include $(BRPC_PATH)/config.mk
+ # Notes on the flags:
+ # 1. Added -fno-omit-frame-pointer: perf/tcmalloc-profiler use frame pointers by default
+-# 2. Added -D__const__= : Avoid over-optimizations of TLS variables by GCC>=4.8
+-CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
++CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
+ ifeq ($(NEED_GPERFTOOLS), 1)
+ 	CXXFLAGS+=-DBRPC_ENABLE_CPU_PROFILER
+ endif
+diff --git a/example/grpc_c++/CMakeLists.txt b/example/grpc_c++/CMakeLists.txt
+index 5f9032e1c..49010e963 100644
+--- a/example/grpc_c++/CMakeLists.txt
++++ b/example/grpc_c++/CMakeLists.txt
+@@ -75,7 +75,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/http_c++/CMakeLists.txt b/example/http_c++/CMakeLists.txt
+index 388b58c2e..34f3050fa 100644
+--- a/example/http_c++/CMakeLists.txt
++++ b/example/http_c++/CMakeLists.txt
+@@ -86,7 +86,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/http_c++/Makefile b/example/http_c++/Makefile
+index dc9d430f6..515456559 100644
+--- a/example/http_c++/Makefile
++++ b/example/http_c++/Makefile
+@@ -20,8 +20,7 @@ BRPC_PATH=../../
+ include $(BRPC_PATH)/config.mk
+ # Notes on the flags:
+ # 1. Added -fno-omit-frame-pointer: perf/tcmalloc-profiler use frame pointers by default
+-# 2. Added -D__const__= : Avoid over-optimizations of TLS variables by GCC>=4.8
+-CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
++CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
+ ifeq ($(NEED_GPERFTOOLS), 1)
+ 	CXXFLAGS+=-DBRPC_ENABLE_CPU_PROFILER
+ endif
+diff --git a/example/memcache_c++/CMakeLists.txt b/example/memcache_c++/CMakeLists.txt
+index 2554b8295..85b4affcb 100644
+--- a/example/memcache_c++/CMakeLists.txt
++++ b/example/memcache_c++/CMakeLists.txt
+@@ -81,7 +81,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/memcache_c++/Makefile b/example/memcache_c++/Makefile
+index 2579f68f4..03b3d4cd7 100644
+--- a/example/memcache_c++/Makefile
++++ b/example/memcache_c++/Makefile
+@@ -17,7 +17,7 @@
+ 
+ BRPC_PATH = ../../
+ include $(BRPC_PATH)/config.mk
+-CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -fPIC -fno-omit-frame-pointer
++CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -fPIC -fno-omit-frame-pointer
+ HDRS+=$(BRPC_PATH)/output/include
+ LIBS+=$(BRPC_PATH)/output/lib
+ HDRPATHS = $(addprefix -I, $(HDRS))
+diff --git a/example/multi_threaded_echo_c++/CMakeLists.txt b/example/multi_threaded_echo_c++/CMakeLists.txt
+index c68010dc1..4a7291c47 100644
+--- a/example/multi_threaded_echo_c++/CMakeLists.txt
++++ b/example/multi_threaded_echo_c++/CMakeLists.txt
+@@ -85,7 +85,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/multi_threaded_echo_c++/Makefile b/example/multi_threaded_echo_c++/Makefile
+index 558b77c36..cd344f98d 100644
+--- a/example/multi_threaded_echo_c++/Makefile
++++ b/example/multi_threaded_echo_c++/Makefile
+@@ -20,8 +20,7 @@ BRPC_PATH=../..
+ include $(BRPC_PATH)/config.mk
+ # Notes on the flags:
+ # 1. Added -fno-omit-frame-pointer: perf/tcmalloc-profiler use frame pointers by default
+-# 2. Added -D__const__= : Avoid over-optimizations of TLS variables by GCC>=4.8
+-CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
++CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
+ ifeq ($(NEED_GPERFTOOLS), 1)
+ 	CXXFLAGS+=-DBRPC_ENABLE_CPU_PROFILER
+ endif
+diff --git a/example/multi_threaded_echo_fns_c++/CMakeLists.txt b/example/multi_threaded_echo_fns_c++/CMakeLists.txt
+index 148490a05..8345076eb 100644
+--- a/example/multi_threaded_echo_fns_c++/CMakeLists.txt
++++ b/example/multi_threaded_echo_fns_c++/CMakeLists.txt
+@@ -85,7 +85,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/nshead_extension_c++/CMakeLists.txt b/example/nshead_extension_c++/CMakeLists.txt
+index 72c9b4183..b0b93a23d 100644
+--- a/example/nshead_extension_c++/CMakeLists.txt
++++ b/example/nshead_extension_c++/CMakeLists.txt
+@@ -81,7 +81,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/nshead_pb_extension_c++/CMakeLists.txt b/example/nshead_pb_extension_c++/CMakeLists.txt
+index 82c055bab..448c7070f 100644
+--- a/example/nshead_pb_extension_c++/CMakeLists.txt
++++ b/example/nshead_pb_extension_c++/CMakeLists.txt
+@@ -81,7 +81,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/parallel_echo_c++/CMakeLists.txt b/example/parallel_echo_c++/CMakeLists.txt
+index 65b251530..b24bb41f9 100644
+--- a/example/parallel_echo_c++/CMakeLists.txt
++++ b/example/parallel_echo_c++/CMakeLists.txt
+@@ -85,7 +85,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/partition_echo_c++/CMakeLists.txt b/example/partition_echo_c++/CMakeLists.txt
+index fecdfa0c3..25d98dbe5 100644
+--- a/example/partition_echo_c++/CMakeLists.txt
++++ b/example/partition_echo_c++/CMakeLists.txt
+@@ -85,7 +85,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/redis_c++/CMakeLists.txt b/example/redis_c++/CMakeLists.txt
+index f1bafece8..a7b008b5a 100644
+--- a/example/redis_c++/CMakeLists.txt
++++ b/example/redis_c++/CMakeLists.txt
+@@ -90,7 +90,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/redis_c++/Makefile b/example/redis_c++/Makefile
+index 4ba9505f0..7c94e195d 100644
+--- a/example/redis_c++/Makefile
++++ b/example/redis_c++/Makefile
+@@ -17,7 +17,7 @@
+ 
+ BRPC_PATH = ../../
+ include $(BRPC_PATH)/config.mk
+-CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -fPIC -fno-omit-frame-pointer
++CXXFLAGS+=$(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -fPIC -fno-omit-frame-pointer
+ HDRS+=$(BRPC_PATH)/output/include
+ LIBS+=$(BRPC_PATH)/output/lib
+ HDRPATHS = $(addprefix -I, $(HDRS))
+diff --git a/example/selective_echo_c++/CMakeLists.txt b/example/selective_echo_c++/CMakeLists.txt
+index 74135282e..7d65c7759 100644
+--- a/example/selective_echo_c++/CMakeLists.txt
++++ b/example/selective_echo_c++/CMakeLists.txt
+@@ -85,7 +85,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/session_data_and_thread_local/CMakeLists.txt b/example/session_data_and_thread_local/CMakeLists.txt
+index 120f92e22..28ba03565 100644
+--- a/example/session_data_and_thread_local/CMakeLists.txt
++++ b/example/session_data_and_thread_local/CMakeLists.txt
+@@ -85,7 +85,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBRPC_ENABLE_CPU_PROFILER")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+diff --git a/example/streaming_echo_c++/CMakeLists.txt b/example/streaming_echo_c++/CMakeLists.txt
+index 57fe261e7..34e041d77 100644
+--- a/example/streaming_echo_c++/CMakeLists.txt
++++ b/example/streaming_echo_c++/CMakeLists.txt
+@@ -81,7 +81,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
++set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -DNDEBUG -O2 -D__const__=__unused__ -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer")
+ 
+ if(CMAKE_VERSION VERSION_LESS "3.1.3")
+     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+diff --git a/example/thrift_extension_c++/Makefile b/example/thrift_extension_c++/Makefile
+index 980b2085b..47b381fdb 100644
+--- a/example/thrift_extension_c++/Makefile
++++ b/example/thrift_extension_c++/Makefile
+@@ -20,8 +20,7 @@ BRPC_PATH = ../../
+ include $(BRPC_PATH)/config.mk
+ # Notes on the flags:
+ # 1. Added -fno-omit-frame-pointer: perf/tcmalloc-profiler use frame pointers by default
+-# 2. Added -D__const__= : Avoid over-optimizations of TLS variables by GCC>=4.8
+-CXXFLAGS = $(CPPFLAGS) -std=c++0x -g -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
++CXXFLAGS = $(CPPFLAGS) -std=c++0x -g -DNDEBUG -O2 -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
+ ifeq ($(NEED_GPERFTOOLS), 1)
+ 	CXXFLAGS+=-DBRPC_ENABLE_CPU_PROFILER
+ endif
+diff --git a/src/butil/errno.h b/src/butil/errno.h
+index 0390a840b..14856eeb6 100644
+--- a/src/butil/errno.h
++++ b/src/butil/errno.h
+@@ -22,7 +22,12 @@
+ #ifndef BUTIL_BAIDU_ERRNO_H
+ #define BUTIL_BAIDU_ERRNO_H
+ 
+-#define __const__
++#ifndef __const__
++// Avoid over-optimizations of TLS variables by GCC>=4.8
++// See: https://github.com/apache/incubator-brpc/issues/1693
++#define __const__ __unused__
++#endif
++
+ #include <errno.h>                           // errno
+ #include "butil/macros.h"                     // BAIDU_CONCAT
+ 
+diff --git a/test/BUILD.bazel b/test/BUILD.bazel
+index c4649ab12..a170eaf8c 100644
+--- a/test/BUILD.bazel
++++ b/test/BUILD.bazel
+@@ -24,7 +24,7 @@ config_setting(
+ COPTS = [
+     "-D__STDC_FORMAT_MACROS",
+     "-DBTHREAD_USE_FAST_PTHREAD_MUTEX",
+-    "-D__const__=",
++    "-D__const__=__unused__",
+     "-D_GNU_SOURCE",
+     "-DUSE_SYMBOLIZE",
+     "-DNO_TCMALLOC",
+diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
+index f3e0c9bb3..980794f5f 100644
+--- a/test/CMakeLists.txt
++++ b/test/CMakeLists.txt
+@@ -52,7 +52,7 @@ else()
+ endif()
+ 
+ set(CMAKE_CPP_FLAGS "${DEFINE_CLOCK_GETTIME} -DBRPC_WITH_GLOG=${WITH_GLOG_VAL} -DGFLAGS_NS=${GFLAGS_NS}")
+-set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} -DBTHREAD_USE_FAST_PTHREAD_MUTEX -D__const__= -D_GNU_SOURCE -DUSE_SYMBOLIZE -DNO_TCMALLOC -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DUNIT_TEST -Dprivate=public -Dprotected=public -DBVAR_NOT_LINK_DEFAULT_VARIABLES -D__STRICT_ANSI__ -include ${PROJECT_SOURCE_DIR}/test/sstream_workaround.h")
++set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} -DBTHREAD_USE_FAST_PTHREAD_MUTEX -D__const__=__unused__ -D_GNU_SOURCE -DUSE_SYMBOLIZE -DNO_TCMALLOC -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DUNIT_TEST -Dprivate=public -Dprotected=public -DBVAR_NOT_LINK_DEFAULT_VARIABLES -D__STRICT_ANSI__ -include ${PROJECT_SOURCE_DIR}/test/sstream_workaround.h")
+ set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -g -O2 -pipe -Wall -W -fPIC -fstrict-aliasing -Wno-invalid-offsetof -Wno-unused-parameter -fno-omit-frame-pointer")
+ use_cxx11()
+ 
+diff --git a/test/Makefile b/test/Makefile
+index 99f7b4f74..5aa90e8be 100644
+--- a/test/Makefile
++++ b/test/Makefile
+@@ -18,7 +18,7 @@
+ NEED_GPERFTOOLS=1
+ NEED_GTEST=1
+ include ../config.mk
+-CPPFLAGS+=-DBTHREAD_USE_FAST_PTHREAD_MUTEX -D__const__= -D_GNU_SOURCE -DUSE_SYMBOLIZE -DNO_TCMALLOC -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DUNIT_TEST -Dprivate=public -Dprotected=public -DBVAR_NOT_LINK_DEFAULT_VARIABLES --include sstream_workaround.h
++CPPFLAGS+=-DBTHREAD_USE_FAST_PTHREAD_MUTEX -D_GNU_SOURCE -DUSE_SYMBOLIZE -DNO_TCMALLOC -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DUNIT_TEST -Dprivate=public -Dprotected=public -DBVAR_NOT_LINK_DEFAULT_VARIABLES --include sstream_workaround.h
+ CXXFLAGS=$(CPPFLAGS) -pipe -Wall -W -fPIC -fstrict-aliasing -Wno-invalid-offsetof -Wno-unused-parameter -fno-omit-frame-pointer -std=c++0x
+ 
+ #required by butil/crc32.cc to boost performance for 10x
+diff --git a/tools/BUILD b/tools/BUILD
+index 99f472fda..04b3abbf9 100644
+--- a/tools/BUILD
++++ b/tools/BUILD
+@@ -16,7 +16,7 @@
+ COPTS = [
+     "-D__STDC_FORMAT_MACROS",
+     "-DBTHREAD_USE_FAST_PTHREAD_MUTEX",
+-    "-D__const__=",
++    "-D__const__=__unused__",
+     "-D_GNU_SOURCE",
+     "-DUSE_SYMBOLIZE",
+     "-DNO_TCMALLOC",
+diff --git a/tools/parallel_http/Makefile b/tools/parallel_http/Makefile
+index 5b8664708..a57c5d5f5 100644
+--- a/tools/parallel_http/Makefile
++++ b/tools/parallel_http/Makefile
+@@ -17,7 +17,7 @@
+ 
+ BRPC_PATH = ../../
+ include $(BRPC_PATH)/config.mk
+-CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -fPIC -fno-omit-frame-pointer -Wno-unused-parameter
++CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -fPIC -fno-omit-frame-pointer -Wno-unused-parameter
+ HDRPATHS = -I$(BRPC_PATH)/output/include $(addprefix -I, $(HDRS))
+ LIBPATHS = -L$(BRPC_PATH)/output/lib $(addprefix -L, $(LIBS))
+ STATIC_LINKINGS += $(BRPC_PATH)/output/lib/libbrpc.a
+diff --git a/tools/rpc_press/Makefile b/tools/rpc_press/Makefile
+index c1d866d05..8cae3033c 100644
+--- a/tools/rpc_press/Makefile
++++ b/tools/rpc_press/Makefile
+@@ -17,7 +17,7 @@
+ 
+ BRPC_PATH = ../../
+ include $(BRPC_PATH)/config.mk
+-CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -fPIC -fno-omit-frame-pointer -Wno-unused-parameter
++CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -fPIC -fno-omit-frame-pointer -Wno-unused-parameter
+ HDRPATHS = -I$(BRPC_PATH)/output/include $(addprefix -I, $(HDRS))
+ LIBPATHS = -L$(BRPC_PATH)/output/lib $(addprefix -L, $(LIBS))
+ STATIC_LINKINGS += $(BRPC_PATH)/output/lib/libbrpc.a
+diff --git a/tools/rpc_replay/Makefile b/tools/rpc_replay/Makefile
+index fcd5eb6f9..8f4eadde8 100644
+--- a/tools/rpc_replay/Makefile
++++ b/tools/rpc_replay/Makefile
+@@ -17,7 +17,7 @@
+ 
+ BRPC_PATH = ../../
+ include $(BRPC_PATH)/config.mk
+-CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -fPIC -fno-omit-frame-pointer -Wno-unused-parameter
++CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -fPIC -fno-omit-frame-pointer -Wno-unused-parameter
+ HDRPATHS = -I$(BRPC_PATH)/output/include $(addprefix -I, $(HDRS))
+ LIBPATHS = -L$(BRPC_PATH)/output/lib $(addprefix -L, $(LIBS))
+ STATIC_LINKINGS += $(BRPC_PATH)/output/lib/libbrpc.a
+diff --git a/tools/rpc_view/Makefile b/tools/rpc_view/Makefile
+index c654cfd5a..13f026cbe 100644
+--- a/tools/rpc_view/Makefile
++++ b/tools/rpc_view/Makefile
+@@ -19,8 +19,7 @@ BRPC_PATH = ../../
+ include $(BRPC_PATH)/config.mk
+ # Notes on the flags:
+ # 1. Added -fno-omit-frame-pointer: perf/tcmalloc-profiler use frame pointers by default
+-# 2. Added -D__const__= : Avoid over-optimizations of TLS variables by GCC>=4.8
+-CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
++CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -Wno-unused-parameter -fPIC -fno-omit-frame-pointer
+ HDRPATHS = -I$(BRPC_PATH)/output/include $(addprefix -I, $(HDRS))
+ LIBPATHS = -L$(BRPC_PATH)/output/lib $(addprefix -L, $(LIBS))
+ STATIC_LINKINGS += $(BRPC_PATH)/output/lib/libbrpc.a
+diff --git a/tools/trackme_server/Makefile b/tools/trackme_server/Makefile
+index 79e8536d0..5bdd53f86 100644
+--- a/tools/trackme_server/Makefile
++++ b/tools/trackme_server/Makefile
+@@ -17,7 +17,7 @@
+ 
+ BRPC_PATH = ../../
+ include $(BRPC_PATH)/config.mk
+-CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -D__const__= -pipe -W -Wall -fPIC -fno-omit-frame-pointer
++CXXFLAGS = $(CPPFLAGS) -std=c++0x -DNDEBUG -O2 -pipe -W -Wall -fPIC -fno-omit-frame-pointer
+ HDRPATHS = -I$(BRPC_PATH)/output/include $(addprefix -I, $(HDRS))
+ LIBPATHS = -L$(BRPC_PATH)/output/lib $(addprefix -L, $(LIBS))
+ STATIC_LINKINGS += $(BRPC_PATH)/output/lib/libbrpc.a

--- a/ports/brpc/fix-boost-ptr.patch
+++ b/ports/brpc/fix-boost-ptr.patch
@@ -1,5 +1,5 @@
 diff --git a/src/brpc/policy/thrift_protocol.cpp b/src/brpc/policy/thrift_protocol.cpp
-index 634f8f37..21b118b1 100755
+index 9871c01..269cfe1 100644
 --- a/src/brpc/policy/thrift_protocol.cpp
 +++ b/src/brpc/policy/thrift_protocol.cpp
 @@ -45,8 +45,7 @@

--- a/ports/brpc/fix-build.patch
+++ b/ports/brpc/fix-build.patch
@@ -1,8 +1,17 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index 057695af..ed4979f6 100644
+index 5f46dc0..fe12819 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -63,7 +63,8 @@ endif()
+@@ -25,7 +25,7 @@ option(WITH_DEBUG_SYMBOLS "With debug symbols" ON)
+ option(WITH_THRIFT "With thrift framed protocol supported" OFF)
+ option(WITH_SNAPPY "With snappy" OFF)
+ option(BUILD_UNIT_TESTS "Whether to build unit tests" OFF)
+-option(BUILD_BRPC_TOOLS "Whether to build brpc tools" ON)
++option(BUILD_BRPC_TOOLS "Whether to build brpc tools" OFF)
+ option(DOWNLOAD_GTEST "Download and build a fresh copy of googletest. Requires Internet access." ON)
+ 
+ # Enable MACOSX_RPATH. Run "cmake --help-policy CMP0042" for policy details.
+@@ -65,7 +65,8 @@ endif()
  
  if(WITH_THRIFT)
      set(THRIFT_CPP_FLAG "-DENABLE_THRIFT_FRAMED_PROTOCOL")
@@ -12,7 +21,7 @@ index 057695af..ed4979f6 100644
  endif()
  
  include(GNUInstallDirs)
-@@ -142,18 +143,21 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+@@ -144,11 +145,14 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
      endif()
  endif()
  
@@ -21,7 +30,7 @@ index 057695af..ed4979f6 100644
 +get_target_property(PROTOBUF_INCLUDE_DIR protobuf::libprotobuf INTERFACE_INCLUDE_DIRECTORIES)
 +set(PROTOBUF_LIBRARIES protobuf::libprotobuf)
  find_package(Threads REQUIRED)
--
+ 
 -find_path(LEVELDB_INCLUDE_PATH NAMES leveldb/db.h)
 -find_library(LEVELDB_LIB NAMES leveldb)
 +find_package(leveldb CONFIG REQUIRED)
@@ -29,6 +38,8 @@ index 057695af..ed4979f6 100644
 +set(LEVELDB_LIB leveldb::leveldb)
  if ((NOT LEVELDB_INCLUDE_PATH) OR (NOT LEVELDB_LIB))
      message(FATAL_ERROR "Fail to find leveldb")
+ endif()
+@@ -163,8 +167,9 @@ if(WITH_SNAPPY)
  endif()
  
  if(WITH_GLOG)
@@ -40,7 +51,7 @@ index 057695af..ed4979f6 100644
      if((NOT GLOG_INCLUDE_PATH) OR (NOT GLOG_LIB))
          message(FATAL_ERROR "Fail to find glog")
      endif()
-@@ -171,7 +175,7 @@ if(WITH_MESALINK)
+@@ -182,7 +187,7 @@ if(WITH_MESALINK)
      include_directories(${MESALINK_INCLUDE_PATH})
  endif()
  
@@ -49,7 +60,7 @@ index 057695af..ed4979f6 100644
  if(NOT PROTOC_LIB)
      message(FATAL_ERROR "Fail to find protoc lib")
  endif()
-@@ -182,7 +186,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+@@ -193,7 +198,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
          )
  endif()
  
@@ -58,7 +69,7 @@ index 057695af..ed4979f6 100644
  
  include_directories(
          ${GFLAGS_INCLUDE_PATH}
-@@ -197,9 +201,9 @@ set(DYNAMIC_LIB
+@@ -208,9 +213,9 @@ set(DYNAMIC_LIB
      ${LEVELDB_LIB}
      ${PROTOC_LIB}
      ${CMAKE_THREAD_LIBS_INIT}
@@ -70,18 +81,8 @@ index 057695af..ed4979f6 100644
      dl
      z)
  
-@@ -434,7 +438,9 @@ if(BUILD_UNIT_TESTS)
-     enable_testing()
-     add_subdirectory(test)
- endif()
-+if(BUILD_TOOLS)
- add_subdirectory(tools)
-+endif()
- 
- file(COPY ${CMAKE_CURRENT_BINARY_DIR}/brpc/
-         DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/output/include/brpc/
 diff --git a/cmake/FindGFLAGS.cmake b/cmake/FindGFLAGS.cmake
-index dfad5fd8..83a167f0 100644
+index dfad5fd..83a167f 100644
 --- a/cmake/FindGFLAGS.cmake
 +++ b/cmake/FindGFLAGS.cmake
 @@ -15,7 +15,8 @@
@@ -104,82 +105,79 @@ index dfad5fd8..83a167f0 100644
    set(GFLAGS_FOUND TRUE)
  endif(GFLAGS_INCLUDE_PATH AND GFLAGS_LIBRARY)
 diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
-index ee616eb1..35a5b48e 100644
+index 209c0e2..b9434ee 100644
 --- a/src/CMakeLists.txt
 +++ b/src/CMakeLists.txt
-@@ -31,45 +31,77 @@ add_dependencies(SOURCES_LIB PROTO_LIB)
+@@ -30,17 +30,22 @@ add_dependencies(SOURCES_LIB PROTO_LIB)
+ # shared library needs POSITION_INDEPENDENT_CODE
  set_property(TARGET ${SOURCES_LIB} PROPERTY POSITION_INDEPENDENT_CODE 1)
  set_property(TARGET ${BUTIL_LIB} PROPERTY POSITION_INDEPENDENT_CODE 1)
- 
--add_library(brpc-shared SHARED $<TARGET_OBJECTS:BUTIL_LIB> 
-+if (BUILD_SHARED_LIBS)
-+add_library(brpc-shared SHARED $<TARGET_OBJECTS:BUTIL_LIB>
-                                $<TARGET_OBJECTS:SOURCES_LIB>
-                                $<TARGET_OBJECTS:PROTO_LIB>)
-+else()
+-
++if(NOT BUILD_SHARED_LIBS)
  add_library(brpc-static STATIC $<TARGET_OBJECTS:BUTIL_LIB>
                                 $<TARGET_OBJECTS:SOURCES_LIB>
                                 $<TARGET_OBJECTS:PROTO_LIB>)
+ 
+-if(BRPC_WITH_THRIFT)
++if(WITH_THRIFT)
+    target_link_libraries(brpc-static thrift)
+ endif()
 -
--target_link_libraries(brpc-shared ${DYNAMIC_LIB})
-+endif()
-+if (BUILD_SHARED_LIBS)
-+target_include_directories(brpc-shared PUBLIC $<INSTALL_INTERFACE:include>)
-+target_link_libraries(brpc-shared PUBLIC ${DYNAMIC_LIB})
-+else()
-+target_include_directories(brpc-static PUBLIC $<INSTALL_INTERFACE:include>)
 +target_link_libraries(brpc-static PUBLIC ${DYNAMIC_LIB})
-+endif()
- 
- if(BRPC_WITH_GLOG)
-+    if (BUILD_SHARED_LIBS)
-     target_link_libraries(brpc-shared ${GLOG_LIB})
-+    else()
++if(BRPC_WITH_GLOG)
 +    target_link_libraries(brpc-static ${GLOG_LIB})
-+    endif()
- endif()
- 
- if(BRPC_WITH_THRIFT)
-+    if (BUILD_SHARED_LIBS)
-     target_link_libraries(brpc-shared thrift)
-+    else()
-     target_link_libraries(brpc-static thrift)
-+    endif()
- endif()
- 
-+if (NOT BUILD_SHARED_LIBS)
- SET_TARGET_PROPERTIES(brpc-static PROPERTIES OUTPUT_NAME brpc CLEAN_DIRECT_OUTPUT 1)
-+else()
- SET_TARGET_PROPERTIES(brpc-shared PROPERTIES OUTPUT_NAME brpc CLEAN_DIRECT_OUTPUT 1)
 +endif()
++target_include_directories(brpc-static PUBLIC $<INSTALL_INTERFACE:include>)
+ SET_TARGET_PROPERTIES(brpc-static PROPERTIES OUTPUT_NAME brpc CLEAN_DIRECT_OUTPUT 1)
+-
++endif()
++if(0)
+ # for protoc-gen-mcpack
+ set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/output/bin)
+     
+@@ -49,35 +54,56 @@ set(protoc_gen_mcpack_SOURCES
+  )
+     
+ add_executable(protoc-gen-mcpack ${protoc_gen_mcpack_SOURCES})
+-
++endif()
+ if(BUILD_SHARED_LIBS)
+     add_library(brpc-shared SHARED $<TARGET_OBJECTS:BUTIL_LIB> 
+                                    $<TARGET_OBJECTS:SOURCES_LIB>
+                                    $<TARGET_OBJECTS:PROTO_LIB>)
+     target_link_libraries(brpc-shared ${DYNAMIC_LIB})
++    target_include_directories(brpc-shared PUBLIC $<INSTALL_INTERFACE:include>)
+     if(BRPC_WITH_GLOG)
+         target_link_libraries(brpc-shared ${GLOG_LIB})
+     endif()
+-    if(BRPC_WITH_THRIFT)
++    if(WITH_THRIFT)
+         target_link_libraries(brpc-shared thrift)
+     endif()
+     SET_TARGET_PROPERTIES(brpc-shared PROPERTIES OUTPUT_NAME brpc CLEAN_DIRECT_OUTPUT 1)
  
--# for protoc-gen-mcpack
--set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/output/bin)
--    
--set(protoc_gen_mcpack_SOURCES
--    ${PROJECT_SOURCE_DIR}/src/mcpack2pb/generator.cpp
-- )
--add_executable(protoc-gen-mcpack ${protoc_gen_mcpack_SOURCES})
--target_link_libraries(protoc-gen-mcpack brpc-shared)
--    
- #install directory
--install(TARGETS brpc-shared
-+if (BUILD_SHARED_LIBS)
-+install(TARGETS brpc-shared EXPORT unofficial-brpcTargets
-         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-         )
+     target_link_libraries(protoc-gen-mcpack brpc-shared ${DYNAMIC_LIB} pthread)
+ 
+-    install(TARGETS brpc-shared
++    install(TARGETS brpc-shared EXPORT unofficial-brpcTargets
+             RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+             )
+ else()
+-    target_link_libraries(protoc-gen-mcpack brpc-static ${BRPC_PRIVATE_LIBS} pthread)
+ endif()
+ 
+ 
 -
 -install(TARGETS brpc-static
-+else()
++if(NOT BUILD_SHARED_LIBS)
 +install(TARGETS brpc-static EXPORT unofficial-brpcTargets
          RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
          LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
          ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
          )
 +endif()
-+
 +file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/unofficial-brpc-config.cmake"
 +[[include(CMakeFindDependencyMacro)
 +find_dependency(OpenSSL)

--- a/ports/brpc/portfile.cmake
+++ b/ports/brpc/portfile.cmake
@@ -7,7 +7,7 @@ vcpkg_from_github(
     PATCHES
         fix-build.patch
         fix-boost-ptr.patch
-        brpc-1783.diff
+        brpc-1783.diff #https://github.com/apache/incubator-brpc/pull/1783
 )
 
 vcpkg_cmake_configure(

--- a/ports/brpc/portfile.cmake
+++ b/ports/brpc/portfile.cmake
@@ -5,7 +5,7 @@ vcpkg_from_github(
     SHA512 bd4c67967796592030903041ddb9205e24c9f196e63ebc153e08fbce723d93d27cd4f30f3c2cf904a93cda66ffa9db7d465d6e5fdac27a045ae84afad3dd1dc3
     HEAD_REF master
     PATCHES
-        #fix-build.patch
+        fix-build.patch
         fix-boost-ptr.patch
         brpc-1783.diff
 )

--- a/ports/brpc/portfile.cmake
+++ b/ports/brpc/portfile.cmake
@@ -1,12 +1,13 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO apache/incubator-brpc
-    REF 06247c18ed6307613f04c0d2357cb91d0c14131b
-    SHA512 f31b1ce3bd99dd585686ec540c9713d5d5936b2d58aadfb3f0ef48faeaab1797f2373e14b73578c2f9f8355d1e9d03661e2ed9ed8c4349b4e43e510d2214b5ae
+    REF 29491107cbf405a494aaf80ee32344ba34e1d7e4 #1.2.0
+    SHA512 bd4c67967796592030903041ddb9205e24c9f196e63ebc153e08fbce723d93d27cd4f30f3c2cf904a93cda66ffa9db7d465d6e5fdac27a045ae84afad3dd1dc3
     HEAD_REF master
     PATCHES
-        fix-build.patch
+        #fix-build.patch
         fix-boost-ptr.patch
+        brpc-1783.diff
 )
 
 vcpkg_cmake_configure(

--- a/ports/brpc/vcpkg.json
+++ b/ports/brpc/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "brpc",
-  "version": "1.1.0",
+  "version": "1.2.0",
   "description": "Industrial-grade RPC framework used throughout Baidu, with 1,000,000+ instances and thousands kinds of services, called \"baidu-rpc\" inside Baidu.",
   "homepage": "https://github.com/apache/incubator-brpc",
   "license": "Apache-2.0",

--- a/versions/b-/brpc.json
+++ b/versions/b-/brpc.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "9d792f86ea59e9347602d6a454b9f3b434b66926",
+      "git-tree": "d75b4f1f3f1bf6072c3095fe34f29e86c19ec4d5",
       "version": "1.2.0",
       "port-version": 0
     },

--- a/versions/b-/brpc.json
+++ b/versions/b-/brpc.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "3637bde5f95f0fd9722caa254f086580012db7d1",
+      "version": "1.2.0",
+      "port-version": 0
+    },
+    {
       "git-tree": "c88bb19658430803981fbfa4ec45432bed88310b",
       "version": "1.1.0",
       "port-version": 0

--- a/versions/b-/brpc.json
+++ b/versions/b-/brpc.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "3637bde5f95f0fd9722caa254f086580012db7d1",
+      "git-tree": "9d792f86ea59e9347602d6a454b9f3b434b66926",
       "version": "1.2.0",
       "port-version": 0
     },

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1165,7 +1165,7 @@
       "port-version": 2
     },
     "brpc": {
-      "baseline": "1.1.0",
+      "baseline": "1.2.0",
       "port-version": 0
     },
     "brunocodutra-metal": {

